bofire/data_models/strategies/predictives/enting.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any, Dict, Literal, Type
+
+from entmoot.models.model_params import EntingParams
+from pydantic import Field
+
+from bofire.data_models.constraints.api import (
+    Constraint,
+    LinearEqualityConstraint,
+    LinearInequalityConstraint,
+    NChooseKConstraint,
+)
+from bofire.data_models.features.api import (
+    CategoricalDescriptorInput,
+    CategoricalInput,
+    ContinuousInput,
+    ContinuousOutput,
+    DiscreteInput,
+    Feature,
+)
+from bofire.data_models.objectives.api import (
+    MaximizeObjective,
+    MinimizeObjective,
+    Objective,
+)
+from bofire.data_models.strategies.predictives.predictive import PredictiveStrategy
+
+
+class EntingStrategy(PredictiveStrategy):
+    type: Literal["EntingStrategy"] = "EntingStrategy"
+    enting_params: EntingParams = Field(default_factory=EntingParams)
```

Comment:
```suggestion
    enting_params: EntingParams = Field(default_factory=lambda: EntingParams())
```

Is this `EntingParams` object something which can be serialized/desirialized by pydantic?

Reply:
Furthermore, we are introducing with this a dependency in the data model to an external library. So far, we tried to avoid this, as we want to be able to perform a minimal installation of the data models without the need for external optimizer libs. Can you wrap this EntingParams thingy for examples as a pydantic class and map it in the functional strategy to the actual object from the entmoot lib? @bertiqwerty what do you think?

Reply:
Agreed. We should keep the data models as self-contained as possible.

Reply:
Hi @jduerholt, could you please clarify this? The entmoot package current has some object that lists all the parameters that it takes. Would you like me to duplicate this list in bofire? This feels like duplication of code which may result in issues (if entmoot gets updated, we would need to update this list in bofire too).

Note the EntingParams is a `dataclass`, so we can just use `pydantic.dataclass`. This means going to/from pydantic will be very easy.

Reply:
I fully agree with @jduerholt. Code duplication sounds evil due to obvious reasons. On the other hand however, code duplication means less dependencies and less coupling which can be more important. As in this case. As Johannes said, since ENTMOOT is not part of BoFire we will need to adapt to changes in the ENTMOOT repo anyway. Our tests should cover all possible parameters so that problems are detected (might be already the case, haven't checked).